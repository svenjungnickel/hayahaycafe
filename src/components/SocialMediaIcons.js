import React from 'react';
import PropTypes from 'prop-types';
import { graphql, StaticQuery } from 'gatsby';
import Button from 'react-bootstrap/Button';
import { FaFacebookF, FaInstagram, FaTripadvisor } from 'react-icons/fa';
import SocialMediaIconsStyles from '../styles/components/SocialMediaIcons.module.scss';

const SocialMediaIcon = ({ item }) => (
    <Button
        href={item.url}
        size="lg"
        target="_blank"
        type="button"
        rel="noreferrer"
        aria-label={`Visit Hayahay Cafe at ${item.type}`}
    >
        {'facebook' === item.type && <FaFacebookF />}
        {'instagram' === item.type && <FaInstagram />}
        {'tripadvisor' === item.type && <FaTripadvisor />}
    </Button>
);

SocialMediaIcon.propTypes = {
    item: PropTypes.shape({
        url: PropTypes.string.isRequired,
        type: PropTypes.string.isRequired,
    }).isRequired,
};

const SocialMediaIcons = () => (
    <StaticQuery
        query={graphql`
            query {
                globalSettings: settingsYaml {
                    socialMedia {
                        url
                        type
                    }
                }
            }
        `}
        render={({ globalSettings }) => (
            <>
                {globalSettings && globalSettings.socialMedia && (
                    <div className={`container text-center ${SocialMediaIconsStyles.socialMediaIcon}`}>
                        {globalSettings.socialMedia.map(item => (
                            <SocialMediaIcon item={item} key={item.type} />
                        ))}
                    </div>
                )}
            </>
        )}
    />
);

export default SocialMediaIcons;
