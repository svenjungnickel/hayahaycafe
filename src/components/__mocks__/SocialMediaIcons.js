import React from 'react';

const SocialMediaIcons = () => (
    <ul>
        <li>facebook</li>
        <li>instagram</li>
        <li>tripadvisor</li>
    </ul>
);

export default SocialMediaIcons;
