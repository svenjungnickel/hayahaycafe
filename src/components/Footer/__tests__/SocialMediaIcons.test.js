import React from 'react';
import renderer from 'react-test-renderer';
import { StaticQuery } from 'gatsby';
import SocialMediaIcons, { SocialMediaIcon } from '../SocialMediaIcons';

describe('SocialMediaIcons', () => {
    it('renders without required item throws prop type error', () => {
        const renderSocialMediaIcon = () => {
            const component = <SocialMediaIcon />;
            renderer.create(component);
        };

        expect(renderSocialMediaIcon).toThrowError('Warning: Failed prop type');
    });

    it('renders without required item url throws prop type error', () => {
        const renderSocialMediaIcon = () => {
            const icon = {
                type: 'type',
            };

            const component = <SocialMediaIcon item={icon} />;
            renderer.create(component);
        };

        expect(renderSocialMediaIcon).toThrowError('Warning: Failed prop type');
    });

    it('renders without required item type throws prop type error', () => {
        const renderSocialMediaIcon = () => {
            const icon = {
                url: 'url',
            };

            const component = <SocialMediaIcon item={icon} />;
            renderer.create(component);
        };

        expect(renderSocialMediaIcon).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid item url throws prop type error', () => {
        const renderSocialMediaIcon = () => {
            const icon = {
                url: 123,
                type: 'facebook',
            };

            const component = <SocialMediaIcon item={icon} />;
            renderer.create(component);
        };

        expect(renderSocialMediaIcon).toThrowError('Warning: Failed prop type');
    });

    it('renders with unknown icon throws prop type error', () => {
        const renderSocialMediaIcon = () => {
            const icon = {
                url: 'url',
                type: 'unknown',
            };

            const component = <SocialMediaIcon item={icon} />;
            renderer.create(component);
        };

        expect(renderSocialMediaIcon).toThrowError('Warning: Failed prop type');
    });

    it('renders facebook icon correctly', () => {
        const icon = {
            url: 'url',
            type: 'facebook',
        };

        const component = <SocialMediaIcon item={icon} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders instagram icon correctly', () => {
        const icon = {
            url: 'url',
            type: 'instagram',
        };

        const component = <SocialMediaIcon item={icon} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders tripadvisor icon correctly', () => {
        const icon = {
            url: 'url',
            type: 'tripadvisor',
        };

        const component = <SocialMediaIcon item={icon} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('does not render empty data', () => {
        const data = {};
        StaticQuery.mockImplementationOnce(({ render }) => render(data));

        const component = <SocialMediaIcons />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('does not render empty globalSettings data', () => {
        const data = {
            globalSettings: {},
        };
        StaticQuery.mockImplementationOnce(({ render }) => render(data));

        const component = <SocialMediaIcons />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('does not render empty socialMedia data array', () => {
        const data = {
            globalSettings: {
                socialMedia: [],
            },
        };
        StaticQuery.mockImplementationOnce(({ render }) => render(data));

        const component = <SocialMediaIcons />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders all icons correctly', () => {
        const data = {
            globalSettings: {
                socialMedia: [
                    {
                        url: 'url',
                        type: 'instagram',
                    },
                    {
                        url: 'url',
                        type: 'facebook',
                    },
                    {
                        url: 'url',
                        type: 'tripadvisor',
                    },
                ],
            },
        };
        StaticQuery.mockImplementationOnce(({ render }) => render(data));

        const component = <SocialMediaIcons />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
