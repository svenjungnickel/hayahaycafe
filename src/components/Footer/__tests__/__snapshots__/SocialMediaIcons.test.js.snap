// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SocialMediaIcons does not render empty data 1`] = `null`;

exports[`SocialMediaIcons does not render empty globalSettings data 1`] = `null`;

exports[`SocialMediaIcons does not render empty socialMedia data array 1`] = `null`;

exports[`SocialMediaIcons renders all icons correctly 1`] = `
<div
  className="container text-center socialMediaIcon"
  data-cy="footerSocialMediaIcons"
>
  <a
    aria-label="Visit Hayahay Cafe at instagram"
    className="btn btn-primary btn-lg"
    disabled={false}
    href="url"
    onClick={[Function]}
    onKeyDown={[Function]}
    rel="noreferrer"
    role="button"
    tabIndex={0}
    target="_blank"
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
      />
    </svg>
  </a>
  <a
    aria-label="Visit Hayahay Cafe at facebook"
    className="btn btn-primary btn-lg"
    disabled={false}
    href="url"
    onClick={[Function]}
    onKeyDown={[Function]}
    rel="noreferrer"
    role="button"
    tabIndex={0}
    target="_blank"
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 320 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
      />
    </svg>
  </a>
  <a
    aria-label="Visit Hayahay Cafe at tripadvisor"
    className="btn btn-primary btn-lg"
    disabled={false}
    href="url"
    onClick={[Function]}
    onKeyDown={[Function]}
    rel="noreferrer"
    role="button"
    tabIndex={0}
    target="_blank"
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 576 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M528.91,178.82,576,127.58H471.66a326.11,326.11,0,0,0-367,0H0l47.09,51.24A143.911,143.911,0,0,0,241.86,390.73L288,440.93l46.11-50.17A143.94,143.94,0,0,0,575.88,285.18h-.03A143.56,143.56,0,0,0,528.91,178.82ZM144.06,382.57a97.39,97.39,0,1,1,97.39-97.39A97.39,97.39,0,0,1,144.06,382.57ZM288,282.37c0-64.09-46.62-119.08-108.09-142.59a281,281,0,0,1,216.17,0C334.61,163.3,288,218.29,288,282.37Zm143.88,100.2h-.01a97.405,97.405,0,1,1,.01,0ZM144.06,234.12h-.01a51.06,51.06,0,1,0,51.06,51.06v-.11A51,51,0,0,0,144.06,234.12Zm287.82,0a51.06,51.06,0,1,0,51.06,51.06A51.06,51.06,0,0,0,431.88,234.12Z"
      />
    </svg>
  </a>
</div>
`;

exports[`SocialMediaIcons renders facebook icon correctly 1`] = `
<a
  aria-label="Visit Hayahay Cafe at facebook"
  className="btn btn-primary btn-lg"
  disabled={false}
  href="url"
  onClick={[Function]}
  onKeyDown={[Function]}
  rel="noreferrer"
  role="button"
  tabIndex={0}
  target="_blank"
  type="button"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      {
        "color": undefined,
      }
    }
    viewBox="0 0 320 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
    />
  </svg>
</a>
`;

exports[`SocialMediaIcons renders instagram icon correctly 1`] = `
<a
  aria-label="Visit Hayahay Cafe at instagram"
  className="btn btn-primary btn-lg"
  disabled={false}
  href="url"
  onClick={[Function]}
  onKeyDown={[Function]}
  rel="noreferrer"
  role="button"
  tabIndex={0}
  target="_blank"
  type="button"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      {
        "color": undefined,
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
    />
  </svg>
</a>
`;

exports[`SocialMediaIcons renders tripadvisor icon correctly 1`] = `
<a
  aria-label="Visit Hayahay Cafe at tripadvisor"
  className="btn btn-primary btn-lg"
  disabled={false}
  href="url"
  onClick={[Function]}
  onKeyDown={[Function]}
  rel="noreferrer"
  role="button"
  tabIndex={0}
  target="_blank"
  type="button"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      {
        "color": undefined,
      }
    }
    viewBox="0 0 576 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M528.91,178.82,576,127.58H471.66a326.11,326.11,0,0,0-367,0H0l47.09,51.24A143.911,143.911,0,0,0,241.86,390.73L288,440.93l46.11-50.17A143.94,143.94,0,0,0,575.88,285.18h-.03A143.56,143.56,0,0,0,528.91,178.82ZM144.06,382.57a97.39,97.39,0,1,1,97.39-97.39A97.39,97.39,0,0,1,144.06,382.57ZM288,282.37c0-64.09-46.62-119.08-108.09-142.59a281,281,0,0,1,216.17,0C334.61,163.3,288,218.29,288,282.37Zm143.88,100.2h-.01a97.405,97.405,0,1,1,.01,0ZM144.06,234.12h-.01a51.06,51.06,0,1,0,51.06,51.06v-.11A51,51,0,0,0,144.06,234.12Zm287.82,0a51.06,51.06,0,1,0,51.06,51.06A51.06,51.06,0,0,0,431.88,234.12Z"
    />
  </svg>
</a>
`;
